src/presetsController.ts

```diff
@@ -171,9 +172,20 @@ export class PresetsController {
         }
 
         presetsFile = await this.validatePresetsFile(presetsFile, file);
-        // Private fields must be set after validation, otherwise validation would fail.
-        this.populatePrivatePresetsFields(presetsFile, file);
-        await this.mergeIncludeFiles(presetsFile, presetsFile, file, referencedFiles);
+
+        if (presetsFile) {
```

Comment:
How does this solve the `include` test?

Reply:
we need to run mergeIncludeFiles to get the original presets file updated with all possible presets. Otherwise, when we try to select a preset and configure, and we try to grab the original preset to do a full expansion, it wouldn't exist. I made a copy by value so it doesn't get expanded after that. Any thoughts?